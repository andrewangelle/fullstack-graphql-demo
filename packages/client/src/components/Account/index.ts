export * from './InventoryList'
export * from './PurchaseList';
export * from './AccountsPage';
export * from './SalesList';
export * from './UserInfo';
export * from './AccountsContainer'
export * from './AccountsPage'

export * from './Register';