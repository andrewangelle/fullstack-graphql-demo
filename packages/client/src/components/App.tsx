import React from 'react';
import { StripeProvider, Elements } from 'react-stripe-elements';

import {
  BcMenu,
  Routes,
} from 'components/'

const Stripe = StripeProvider as any

export function App(props: any) {
  return (
    <Stripe apiKey={process.env.REACT_APP_STRIPE_TOKEN}>
      <Elements>
        <BcMenu {...props}>
          <Routes {...props} />
        </BcMenu>
      </Elements>
    </Stripe>
  )
}
