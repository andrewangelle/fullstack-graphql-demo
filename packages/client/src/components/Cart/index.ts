export * from './PaymentPanel'
export * from './CartItems';
export * from './ItemsTable';
export * from './ItemQuantity';
export * from './RemoveIcon';
export * from './AddItemButton';
export * from './CartPage';
