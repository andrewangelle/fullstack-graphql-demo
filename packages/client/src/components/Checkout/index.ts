export * from './Payment';
export * from './CheckoutPage';

export * from './ShippingForm';
export * from './PaymentForm';
export * from './AuthorizeForm';