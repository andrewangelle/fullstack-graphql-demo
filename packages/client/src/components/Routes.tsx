
import React from 'react';
import { Router } from '@reach/router';

import {
  AccountsPage,
  CheckoutPage,
  LandingPage,
  LoginPage,
  LogoutPage,
  RegisterPage,
  ShopPage,
  ProductPage,
} from 'components/'

function Routes(props: any) {
  return (
    <Router>
      <LandingPage path="/" {...props} />
      <LoginPage path="/login" {...props} />
      <LogoutPage path="/logout" {...props} />
      <ProductPage path="/product" {...props} />
      <RegisterPage path="/register" {...props} />
      <AccountsPage path="/account" {...props} />
      <CheckoutPage path="/checkout" {...props} />
      <ShopPage path="/shop" {...props} />
    </Router>
  )
}

export { Routes }
