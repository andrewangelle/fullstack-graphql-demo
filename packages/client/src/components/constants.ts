export const wrapperStyles = {
  width: '50%',
  marginLeft: 'auto',
  marginRight: 'auto'
}

export const centerStyles = props => ({
  width: props.width || '50%',
  marginLeft: 'auto',
  marginRight: 'auto'
}) 

export const contentWrapperStyles = {
  margin: '24px 16px', 
  padding: 24, 
  background: '#fff', 
  minHeight: 800 
}

export const lightBlue = {color: '#1890ff'}

export const formItemLayout = {
  labelCol: { span: 4 },
  wrapperCol: { span: 14 },
}

export const pointer = {
  cursor: 'pointer'
}
export const flex = {
  display: 'inline-flex'
}
export const spacer = {
  margin: 'auto 0.5rem'
}


export const navMenu = {
  margin: ' 0 auto',
  display: 'flex',
  justifyContent: 'space-between',
  border: 'none'
}