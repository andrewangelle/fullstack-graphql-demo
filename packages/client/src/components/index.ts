export * from './constants';
export * from './shared';
export * from './Checkout';
export * from './Account';
export * from './Cart';
export * from './Shop'
export * from './Main';

export * from './Login/LoginForm';
export * from './Login/LoginPage';
export * from './Login/LogoutPage';

export * from './App'
export * from './Root'
export * from './Routes'