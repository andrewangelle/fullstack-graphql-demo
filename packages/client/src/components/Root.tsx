import React from 'react';
import { ApolloClient, NormalizedCacheObject } from 'apollo-boost';
import { ApolloProvider } from 'react-apollo';

import { App, Loader } from 'components/';
import { createClient } from 'store/'
import 'styles/index.css';

interface RootState {
  loaded: boolean;
  client: ApolloClient<NormalizedCacheObject> | null;
}

class Root extends React.Component<{}, RootState> {
  state = {
    loaded: false,
    client: null,
  }

  async componentDidMount() {
    const client = await createClient();
    await (window as any).Stripe
    this.setState({ client })
  }

  componentDidUpdate(prevState: RootState) {
    if (prevState.client !== this.state.client) {
      if (this.state.client && !this.state.loaded) {
        this.setState({ loaded: true })
      }
    }
  }

  render() {
    const { loaded, client } = this.state;

    if (!loaded) {
      return <Loader />
    }

    return client !== null && loaded && (
      <ApolloProvider client={client}>
        <App {...this.state} />
      </ApolloProvider>
    )

  }
}


export { Root }