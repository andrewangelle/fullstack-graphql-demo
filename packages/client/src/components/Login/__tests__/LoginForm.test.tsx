import React from 'react';
import { InMemoryCache } from 'apollo-boost';
import wait from 'waait';

import { LoginForm } from 'components/';
import { mockApollo, loginMock } from 'test-utils';

const original = console.error

describe('LoginForm', () => {
  beforeAll(() => console.error = () => '')
  afterAll(() => { console.error = original })

  const component = mockApollo(<LoginForm />, {
    mocks: [loginMock],
    cache: new InMemoryCache()
  });

  it('renders login', async () => {
    await wait(0)
    expect(component.toJSON()).toMatchSnapshot();
  })

  it('sets inputs', async () => {
    await wait(0)

    const email = component.root.find(el => el.props.name === 'email')
    const password = component.root.find(el => el.props.name === 'password')

    email.props.onChange({ target: { value: 'coffeecustomer@gmail.com' } });
    expect(email.props.value).toBe('coffeecustomer@gmail.com');

    password.props.onChange({ target: { value: 'test' } });
    expect(password.props.value).toBe('test')
  })
})
