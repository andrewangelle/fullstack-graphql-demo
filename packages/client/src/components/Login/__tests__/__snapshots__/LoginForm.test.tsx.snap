// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm renders login 1`] = `
<form
  className="ant-form ant-form-horizontal"
  onSubmit={[Function]}
>
  <div
    className="ant-row ant-form-item"
  >
    <div
      className="ant-form-item-label"
    >
      <label
        className=""
        htmlFor={null}
        onClick={[Function]}
        title="email"
      >
        email
      </label>
    </div>
    <div
      className="ant-form-item-control-wrapper"
    >
      <div
        className="ant-form-item-control"
      >
        <span
          className="ant-form-item-children"
        >
          <input
            className="ant-input"
            disabled={false}
            name="email"
            onChange={[Function]}
            onKeyDown={[Function]}
            title="email"
            type="text"
            value=""
          />
        </span>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
  >
    <div
      className="ant-form-item-label"
    >
      <label
        className=""
        htmlFor={null}
        onClick={[Function]}
        title="password"
      >
        password
      </label>
    </div>
    <div
      className="ant-form-item-control-wrapper"
    >
      <div
        className="ant-form-item-control"
      >
        <span
          className="ant-form-item-children"
        >
          <input
            className="ant-input"
            disabled={false}
            name="password"
            onChange={[Function]}
            onKeyDown={[Function]}
            title="password"
            type="text"
            value=""
          />
        </span>
      </div>
    </div>
  </div>
  <button
    className="ant-btn ant-btn-primary"
    onClick={[Function]}
    style={
      Object {
        "marginLeft": "25%",
        "marginRight": "25%",
        "width": "50%",
      }
    }
    type="button"
  >
    <span>
      Login
    </span>
  </button>
</form>
`;
